src/main/java/com/ericsson/eiffel/remrem/semantics/validator/EiffelValidator.java

```diff
@@ -72,41 +88,133 @@ public EiffelValidator(String schemaResourceName, String eventType, List<String>
             throw new IllegalArgumentException(message, e);
         }
     }
-
-    public void validate(JsonObject jsonObjectInput) throws EiffelValidationException {
+    public JsonObject validate(JsonObject jsonObjectInput) throws EiffelValidationException {
+        return validate(jsonObjectInput, false);
+    }
+    
+    public JsonObject validate(JsonObject jsonObjectInput, Boolean lenientValidation) throws EiffelValidationException {
+        JsonArray remremGenerateFailures = new JsonArray();
+        log.debug("VALIDATING Schema used: {}", schemaResourceName);
         try {
-            ProcessingReport report = validationSchema.validate(JsonLoader.fromString(jsonObjectInput.toString()));
-            if (!report.isSuccess()) {
-                log.warn(jsonObjectInput.toString());
-                throw new EiffelValidationException(getErrorsList(report));
+            final ProcessingReport report = validationSchema.validate(JsonLoader.fromString(jsonObjectInput.toString()),
+                    true);
+            if (!report.isSuccess() && lenientValidation) {
+                log.info("Trying to revalidate the Eiffel message with only mandatory Eiffel message fields.");
+                String revalidatedJson = removeErrorProperties(report, jsonObjectInput, remremGenerateFailures);
+                ProcessingReport report2 = validationSchema.validate(JsonLoader.fromString(revalidatedJson), true);
+                handleErrorReport(jsonObjectInput, report2);
+                log.debug("VALIDATED. Schema used: {}", schemaResourceName);
+                return addRemremGenerateFailuresToCustomData(new JsonParser().parse(revalidatedJson).getAsJsonObject(), remremGenerateFailures);
+            } else {
+                handleErrorReport(jsonObjectInput, report);
             }
             log.debug("VALIDATED. Schema used: {}", schemaResourceName);
+            return jsonObjectInput;
         } catch (Exception e) {
-            String message = "Cannot validate given JSON string";
+            final String message = "Cannot validate given JSON string";
             log.debug(message, e);
             throw new EiffelValidationException(message, e);
         }
     }
+    private void handleErrorReport(JsonObject jsonObjectInput, ProcessingReport report) throws EiffelValidationException {
```

Comment:
Blank line between methods please (here and in several other places in the file).

Reply:
done